packages/expo-modules-autolinking/CHANGELOG.md

```diff
@@ -11,6 +11,8 @@
 - Add `--source-dir` option ([#38218](https://github.com/expo/expo/pull/38218) by [@gabrieldonadel](https://github.com/gabrieldonadel))
 - [iOS] Add `:projectRoot` option to `use_expo_modules!` ([#38210](https://github.com/expo/expo/pull/38210) by [@gabrieldonadel](https://github.com/gabrieldonadel))
 - [Android] Support auto-linking C++-only Turbo Modules on Android (as per `@react-native-community/cli-platform-android` changes) ([#38626](https://github.com/expo/expo/pull/38626) by [@kitten](https://github.com/kitten))
+- Respect `expo.autolinking.exclude` option for React Native modules as well ([#38635](https://github.com/expo/expo/pull/38635) by [@kitten](https://github.com/kitten))
```

Comment:
I don't know how I feel about this change. I'm okay with having an option to exclude RN modules. However, I'm not sure if we should merge it with the option for Expo modules autolinking. Additionally, RN modules are linked using a different command, and someone might still use the community CLI to link TurboModules. I don't have a strong opinion, but I would prefer adding it as a separate field or even a separate object called `rn_autolinking`.

Reply:
Discussed offline: Currently, `exclude` is basically an exception, in that, other options from `expo.autolinking` applied to `react-native-config` too but not `exclude`. The expectation and idea here is that `autolinking` is first and foremost about Node module discovery, so aligning is important to keep documentation simple and intuitive.

But if we discover issues with this during the beta, I'd revisit this and potentially revert it or separate this option out.